src/viam/app/data_client.py

```diff
@@ -288,14 +289,18 @@ async def tabular_data_by_mql(self, organization_id: str, mql_binary: List[bytes
         Args:
             organization_id (str): The ID of the organization that owns the data.
                 You can obtain your organization ID from the Viam app's organization settings page.
-            mql_binary (List[bytes]): The MQL query to run as a list of BSON queries. You can encode your bson queries using a library like
-                `pymongo`.
+            mql_queries (Union[List[bytes], List[Dict[str, Union[ValueTypes, datetime]]]]): The MQL query to run as a list of BSON queries.
+                Note: Support for bytes will be removed in the future, so using a dictionary is preferred.
 
         Returns:
             List[Dict[str, Union[ValueTypes, datetime]]]: An array of decoded BSON data objects.
 
         For more information, see `Data Client API <https://docs.viam.com/appendix/apis/data-client/>`_.
         """
+        if isinstance(mql_queries[0], dict):
+            mql_binary = [bson.encode(query) for query in mql_queries if isinstance(query, dict)]
+        else:
+            mql_binary = [query for query in mql_queries if isinstance(query, bytes)]
```

Comment:
I wonder if it would make sense to raise in a case where the user does a mix and match? Right now, if e.g. someone had a list with 1 dict followed by 100 bytes (or vice versa), we'd only capture the first item and drop everything else. But, it seems to me that we would consider the user input to be malformed in such a case. Raising or returning an error to clarify that we need _all_ bytes or _all_ dicts seems better than silently ignoring certain inputs, imo.

Reply:
You're completely right. I was trying to appease the linter but in the process ignored errors completely, which we definitely DON'T want. I think this is an appropriate place to ignore the linter as an error will pop up here if it's not of the correct type.